### PR TITLE
chore: allow CI to run in forks

### PR DESCRIPTION
**Pull request by justplaz**
*Issue #, if available:*

*Description of changes:*

Previously, the CI used hard-coded AWS resource identifiers to run against, which makes running CI in a fork of the repo functionally impossible. 
This change moves the CI to use the repository's Action variables to specify the resources, so that forks can configure their own AWS resources. 
This _does not_ allow execution of 3p PRs against the upstream repo when the submitting fork has not gone through the setup process. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by justplaz** [comment]
It turns out that I missed an important detail about Github repo variables:

> They are not passed to workflows that are triggered by a pull request from a fork.

Therefore, this PR does not allow for PRs from forks against upstream to run CI; no variables are used in this case, so no resources are specified and the tests cannot run. It would be possible to define our (AWS Crypto Tools owned) resources through default values (either in the yaml files somehow, or simply in the Java code) but the STS call will use the fork in the `userIdentity` struct and thus give `AccessDenied` as the role's trust policy specifies the `aws` org's repo. 

This PR is still useful, as it allows for forks to define their own variables to specify AWS resources to test against. That at least lets the 3rd party run CI in their own fork before submitting a PR.

**Comment by justplaz** [comment]
Reviewers, please reference the passing CI in my fork: https://github.com/justplaz/amazon-s3-encryption-client-java/actions/runs/5316565223

Since I am not actually a third party, I have also pushed the branch directly to upstream (this) repo: https://github.com/aws/amazon-s3-encryption-client-java/actions/runs/5327696123

File: README.md [hunk 1]
```diff
@@ -8,10 +8,30 @@ for how to use this library, refer to the
...
 * `AWS_S3EC_TEST_BUCKET` - The bucket to write test values to
-* `AWS_S3EC_TEST_KMS_KEY_ID` - The key id for the KMS key used for KMS tests
-* `AWS_S3EC_TEST_KMS_KEY_ALIAS` - An alias for the KMS key used for KMS tests. The alias must reference the key ID above. 
+* `AWS_S3EC_TEST_KMS_KEY_ID` - The key ID for the KMS key used for KMS tests
+* `AWS_S3EC_TEST_KMS_KEY_ALIAS` - An alias for the KMS key used for KMS tests. The alias must reference the key ID above.
 * `AWS_REGION` - The region the AWS resources (KMS key, S3 bucket) resides e.g. "us-east-1"
 
+To create these resources, refer to the included CloudFormation template (cfn/S3EC-GitHub-CF-Template).
+Make sure that the repo in the trust policy of the IAM role refers to your fork instead of the `aws` organization.
+Note that your account may incur charges based on the usage of any resources beyond the AWS Free Tier.
```
Comment (josecorella) [inline README.md:17]: nit:
```suggestion
**NOTE**: Your account may incur charges based on the usage of any resources beyond the AWS Free Tier.
```

File: README.md [hunk 2]
```diff
@@ -8,10 +8,30 @@ for how to use this library, refer to the
...
+
+If you have forked this repo, there are additional steps required.
+You will need to configure your fork's Github Actions settings to be able to run CI:
+
+Under Settings -> Actions -> General -> Workflow permissions, ensure Read and write permissions is selected.
```
Comment (josecorella) [inline README.md:22]:
```suggestion
Under Settings -> Actions -> General -> Workflow permissions, ensure Read and Write permissions are selected.
```

**Review by josecorella — DISMISSED** [review]
just some small nits, but lgtm

File: README.md (hunk 2 above)
Comment (justplaz) [reply README.md:22]: The more correct fix, IMO, is to put "Read and write permissions" in quotes, as this refers to the singular option in its original casing.

**Comment by justplaz** [comment]
Passing CI in my fork: https://github.com/justplaz/amazon-s3-encryption-client-java/actions/runs/5360137602

**Review by josecorella — APPROVED** [review]

**Pull request merged**